### DISCUSSION TITLE
Echarts native support for accessibility

### DISCUSSION THREAD

**Discussion by frassinier** [apache/superset#32740] [Ideas]
## Screenshot

<img width="1676" alt="image" src="https://github.com/apache/superset/assets/18534166/61143b5a-ecd4-4070-ab54-a536e74ed5ac">

## Description

>After setting [aria.show](https://echarts.apache.org/option.html#aria.show) to true, ECharts will automatically generate a description of the chart according to the title, chart, data, etc. Users can also set description manually through the configuration object.

Increasing accessibility support in Apache/Superset through enhanced a11y features, such as generating [aria-label with Echarts](https://echarts.apache.org/handbook/en/best-practices/aria/), is crucial for ensuring inclusivity and usability for all users, including those with disabilities.

By adhering to [WCAG guidelines](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Use_of_color) and making adjustments to colors and patterns as required, we not only enhance the user experience for individuals with disabilities but also align with standards for accessibility, promoting a more inclusive and equitable platform for all users.

[Apache/Echarts](https://github.com/apache/echarts) has built-in capabilities for describing charts. 
Let's decide to enable this feature by default, promoting inclusive content for all users. Otherwise, it may be worth considering the implementation of a feature flag to provide flexibility and accommodate varying user preferences.
 
## Design input

If we wish to take additional steps, we can also consider altering the pattern of each series, as illustrated in https://github.com/apache/superset/pull/28171 description

**Comment by villebro** [comment]
@frassinier great feature request/POC! This is something we're eagerly hoping to support soon, but we need a mechanism for making it possible to enable/disable this type of functionality on a per-user basis. Please take a look at #28047 for more details.

**Comment by frassinier** [comment]
@villebro Excellent! I like the idea of setting up a user preferences service!

**Comment by rusackas** [comment]
Hah, just left a comment to that effect on the PR, but it looks like you beat me to the punch here :)
